### DISCUSSION TITLE
TypeConverter and validate order

### DISCUSSION THREAD

**Discussion by d-lepestipon** [JoshClose/CsvHelper#1639] [Q&A]
Hi,
I m facing with a common use case, but ca'nt handle it correctly.
I must remove some special chars from a field  and AFTER validate it.

I use a TypeConverter and the Validate method. But Validate triggers before type convertion.
Is there a way to change this?

```csharp
Map(m => m.NroFttOoH).Index(5).TypeConverter<NormalizedStringConverter>().Validate(field => !field.Any(illegalChars));
```
It should be possible to do it, I think my use case is very common.

I know I could use ConvertUsing, but this method is too generic, using it for evry field break the readability. As it take the full row, "strange code" come :

```csharp
Map(m => m.CodeInsee).Name("CD_INSEE", "CODE_INSEE").ConvertUsing((f) => f.GetField(3).RemoveSpecialChars());
```
My header can change, but I can't reflect it in `GetField` (only one string allowed), so I must lock the index. So now index of these column (even with header) is fixed.

I think `ConvertUsing` should be used less possible, and use separated fluent methods instead.

**Comment by JoshClose** [comment]
Validate happens on the raw field and I think that's how that should be. Most cases will be non strings after a conversion, so validation doesn't make sense to me in that case.

Can you pull out the code from `NormalizedStringConverter` into a common place, and run it in the `Validate` method?

The new parser I'm working on will have a configuration methods to modify the field during parsing, so you could probably take advantage of that when it's released.
